pkg/catalog/catalog.go

```diff
@@ -204,6 +212,10 @@ func New(ctx context.Context, cfg Config) (*Catalog, error) {
 	refManager := ref.NewKVRefManager(executor, *cfg.KVStore, ident.NewHexAddressProvider(), nil)
 	gcManager := retention.NewGarbageCollectionManager(tierFSParams.Adapter, refManager, cfg.Config.GetCommittedBlockStoragePrefix())
 	settingManager := settings.NewManager(refManager, *cfg.KVStore)
+	if cfg.CacheFactory != nil {
```

Comment:
Prefer the default to use the default cache and override the cache settings in the specific test.

Reply:
Notice that there's no dependency injection to the `Catalog`. It creates its dependencies. Tests can't override the settings since they do not create the `SettingsManager`

Reply:
Prefer we keep passing options (WithStagingCache) that can override the defaults.
I also think that having Config here is a shortcut.